Client/lib/gMap.js

```diff
@@ -0,0 +1,231 @@
+
+var gMap = function(){
+
+  getGeolocation(startGMap);
+  
+  var lastLocationClicked;
+  var markerCords = [];
+  var travelButtonList = [];
+  var travelMode = 'WALKING';
+  var map;
+
+  function startGMap(pos){
+    //can be -> BICYCLING, DRIVING, TRANSIT, WALKING
+
+    //----------------------------------
+    //creatig the Map
+    //----------------------------------
+    var mapOptions = {
+      center: pos,
+      zoom: 12,
+      backgroundColor: '#568',
+      disableDoubleClickZoom : true,
+      disableDefaultUI: true,
+      keyboardShortcuts: false
+    };
+  	map = new google.maps.Map(document.getElementById('gMap'),mapOptions);
+    
+    //----------------------------------
+    //styling the Map
+    //----------------------------------
+
+    var mapStyles = [
+      {
+        stylers: [ { hue: "#00ffe6" }, { saturation: -20 } ]
+      },{
+        featureType: "road",
+        elementType: "geometry",
+        stylers: [ { lightness: 100 }, { visibility: "simplified" } ]
+      },{
+        featureType: "road",
+        elementType: "labels",
+        stylers: [ { visibility: "off" } ]
+      }
+    ];
+
+    var styledMap = new google.maps.StyledMapType(mapStyles,{name: "Styled Map"});
+    map.mapTypes.set('map_style', styledMap);
+    map.setMapTypeId('map_style');
+
+    //----------------------------------
+    //events
+    //----------------------------------
+    google.maps.event.addListener(map, 'click', function(event) {
+      addMarker(event.latLng, 'Count ' + markerCords.length);
```

Comment:
I imagine this will end up passing an Object back to the front end for them to work with. If so consider adding a comment in code that describes this object such as 
{
  Lat: Number
  Lon: Number
ect...